src/components/users/List.vue

```diff
@@ -0,0 +1,188 @@
+<script setup lang="ts">
+import { useAsyncState } from '@vueuse/core'
+import { ref, Ref } from 'vue'
+import { useAuth } from '../../composables/useAuth'
+import { useSessionToken } from '../../composables/useSessionToken'
+import { Users } from '../../lib/api/Users'
+import Paginator from '../leaderboards/Paginator.vue'
+import { UserRole } from '../../lib/api/data-contracts'
+
+const props = defineProps<{
+	limit: number | undefined
+	page: number
+	roles: string | string[] | undefined
+}>()
+
+function parseRolesFromQuery(roles: string | string[] | undefined): UserRole[] {
+	if (roles === undefined) {
+		return []
+	}
+
+	if (typeof roles === 'string') {
+		if (/^administrator$/i.test(roles)) {
+			return ['Administrator']
+		}
+		if (/^banned$/i.test(roles)) {
+			return ['Banned']
+		}
+		if (/^confirmed$/i.test(roles)) {
+			return ['Confirmed']
+		}
+		if (/^registered$/i.test(roles)) {
+			return ['Registered']
+		}
+		return []
+	}
+
+	const r: Set<UserRole> = new Set()
+
+	for (let role of roles) {
+		if (/^administrator$/i.test(role)) {
+			r.add('Administrator')
+			continue
+		}
+		if (/^banned$/i.test(role)) {
+			r.add('Banned')
+			continue
+		}
+		if (/^confirmed$/i.test(role)) {
+			r.add('Confirmed')
+			continue
+		}
+		if (/^registered$/i.test(role)) {
+			r.add('Registered')
+			continue
+		}
+	}
```

Comment:
same here.

Reply:
We don't have this block anymore; the refactor seen here https://github.com/leaderboardsgg/leaderboard-admin/pull/48/files#diff-c8875609761701e9d75a84363168a69daaeaf6bc75d9ca9e16b2c11484981c04R14-R27 now calls `useRouteQuery` and does an array `filter` on the query against the set of valid roles. Refer to the linked commit block for more context.